### PR TITLE
api: major spring cleaning

### PR DESCRIPTION
**Pull request by wukko**

### CODE REVIEW

File: src/core/api.js [hunk 1]
```diff
@@ -63,136 +63,152 @@ export function runAPI(express, app, gitCommit, gitBranch, __dirname) {
...
         next();
-    });
+    })
 
     app.use('/api/json', express.json({
         verify: (req, res, buf) => {
-            let acceptCon = String(req.header('Accept')) === "application/json";
-            if (acceptCon) {
+            const acceptHeader = String(req.header('Accept')) === "application/json";
+            if (acceptHeader) {
```
Comment (lostdusty) [inline src/core/api.js:71]: Couldnt this be a single if?

**Review by lostdusty — COMMENTED** [review]
Is there a beta instance where we can test?

**Comment by hyperdefined** [comment]
I made a test instance if anyone wants to test it here: https://cobalt-test.canine.tools/ (api only)

**Review by dumbmoron — APPROVED** [review]

**Review by synzr — COMMENTED** [review]
lgtm

File: src/modules/processing/services/soundcloud.js [hunk 2]
```diff
@@ -75,12 +75,15 @@ export default async function(obj) {
     let fileUrlBase = selectedStream.url;
     let fileUrl = `${fileUrlBase}${fileUrlBase.includes("?") ? "&" : "?"}client_id=${clientId}&track_authorization=${json.track_authorization}`;
 
-    if (fileUrl.substring(0, 54) !== "https://api-v2.soundcloud.com/media/soundcloud:tracks:") return { error: 'ErrorEmptyDownload' };
+    if (fileUrl.substring(0, 54) !== "https://api-v2.soundcloud.com/media/soundcloud:tracks:")
```
Comment (dumbmoron) [inline src/modules/processing/services/soundcloud.js:78]: can we finally get rid of this and just use .startsWith() please please please

Comment (wukko) [reply src/modules/processing/services/soundcloud.js:78]: ok 👍

**Review by dumbmoron — APPROVED** [review]

**Comment by wukko** [comment]
lgtv

**Comment by dumbmoron** [comment]
image

**Comment by lostdusty** [comment]
😭

**Pull request merged**